browser_patches/firefox-beta/juggler/NetworkObserver.js

```diff
@@ -133,6 +132,17 @@ class NetworkRequest {
       // use onStopRequest, but that will only happen after the last redirect has finished.
       redirectedFrom._sendOnRequestFinished();
     }
+    // In case of proxy auth, we get to requests with the same channel:
+    // - one is pre-auth
+    // - second is with auth header.
+    //
+    // In this case, we create this NetworkRequest object with a `redirectedFrom`
+    // object, and they both share the same httpChannel.
+    //
+    // Since we want to maintain _channelToRequest map without clashes,
+    // we must call `_sendOnRequestFinished` **before** we update it with a new object
+    // here.
+    this._networkObserver._channelToRequest.set(this.httpChannel, this);
```

Comment:
Let's add an assert that the channel is not in the map (already).

Reply:
done